app/api/api_v1/endpoints/data.py

```diff
@@ -0,0 +1,83 @@
+import json
+
+from fastapi import APIRouter, File, Depends, HTTPException, UploadFile
+from sqlalchemy.orm import Session
+
+from api import deps
+from models import User
+
+import crud
+
+from schemas import DataCreate, DataID, Message
+
+router = APIRouter()
+
+
+@router.get("/{dataset_id}")
+def get_by_id(
+        dataset_id: int,
+        current_user: User = Depends(deps.get_current_user),
```

Comment:
why do you requiere `current_user` ?

Reply:
Authentication, if the user is not logged in, then they won't have access to the api